src/main/java/com/code_intelligence/jazzer/mutation/mutator/collection/MapMutatorFactory.java

```diff
@@ -126,32 +130,105 @@ public void initInPlace(Map<K, V> reference, PseudoRandom prng) {
       }
     }
 
+    private void eraseRandomChunk(Map<K, V> map, PseudoRandom prng, int mapSize) {
+      int upperLimit = Math.max(1, Math.min(mapSize - minSize, mapSize / 2));
+      int chunkSize = upperLimit == 1 ? 1 : prng.closedRange(1, upperLimit);
+      int chunkOffset = prng.closedRange(0, mapSize - chunkSize);
+      int elementsRemoved = 0;
+      while (elementsRemoved < chunkSize) {
+        map.entrySet()
```

Comment:
This iterates over the map in every iteration. Instead, you could collect the keys to remove in a temporary list and then remove them via `map.keySet().removeAll()`.